### PR TITLE
Module users and rights

### PR DESCRIPTION
**Pull request by Grenadingue**

### CODE REVIEW

File: tmp/#test# [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { createStore } from 'redux'
```
Comment (clairealmozinos) [inline tmp/#test#]: @leroy-p what is this file?

File: tmp/.#test [hunk 2]
```diff
@@ -0,0 +1 @@
+etienne@PC-Etienne.3445:1466695986
```
Comment (clairealmozinos) [inline tmp/.#test]: @leroy-p what is this file?

File: views/users_and_rights/login_signup.ejs [hunk 3]
```diff
@@ -43,5 +43,16 @@
             <button type="submit">Sign-up</button>
```
Comment (clairealmozinos) [inline views/users_and_rights/login_signup.ejs]: @Grenadingue why is there still some ejs file?

File: webapp/app/components/EditUser/index.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+EditUser.propTypes = {
+  state: React.PropTypes.object,
+  onChangeData: React.PropTypes.func,
+  getUserRequest: React.PropTypes.func,
+  getCurrentUserRequest: React.PropTypes.func,
+  editUserRequest: React.PropTypes.func,
+};
+
+// let adminField;
```
Comment (clairealmozinos) [inline webapp/app/components/EditUser/index.js:76]: Test commented code should be remove (After merge).

File: webapp/app/components/Navbar/index.js [hunk 5]
```diff
@@ -1,16 +1,13 @@
 //
```
Comment (clairealmozinos) [inline webapp/app/components/Navbar/index.js:1]: @leroy-p I asked you to not commit this file (Before merge).

File: webapp/app/containers/Users/actions.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    users: users,
+  }
+}
+
+export function getUsersRequest() {
+  return function returnGetUsersRequest(dispatch) {
+    return request
+      .get('http://localhost:8080/api/logged-in/admin/users')
+      .end((err, res) => {
+        console.log(res.body);
```
Comment (clairealmozinos) [inline webapp/app/containers/Users/actions.js]: @leroy-p console.log should not be commit.

File: managers/users.js [hunk 7]
```diff
@@ -67,142 +72,291 @@ function checkAndCreateUser(name, email, password, confirmation) {
...
+  } else {
+    next();
+  }
+};
+
+//
+// Security check for each url of this kind : /api/logged-in/*
+//
+module.exports.ensureLoggedIn = function (req, res, next) {
+  console.log(req.isAuthenticated());
```
Comment (clairealmozinos) [inline managers/users.js]: @Grenadingue console.log should not be commit.

File: webapp/app/containers/EditGroup/actions.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    displayedSoftwarePackagesMode: group.softwarePackagesMode,
+  }
+}
+
+export function getGroupRequest(groupname) {
+  return function returnGetGroupRequest(dispatch) {
+    return request
+      .get('http://localhost:8080/api/logged-in/admin/groups')
+      .end((err, res) => {
+        console.log(res.body.groups[1].saveAndRestoreMode);
```
Comment (clairealmozinos) [inline webapp/app/containers/EditGroup/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/EditGroup/actions.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        var i = 0;
+        while (i < res.body.groups.length && res.body.groups[i].name != groupname) {
+          ++i;
+        }
+        console.log('Group displayed : ' + res.body.groups[i].name);
```
Comment (clairealmozinos) [inline webapp/app/containers/EditGroup/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/EditGroup/actions.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+export function editGroup(group) {
+  return {
+    type: EDIT_GROUP,
+    group: group,
+  };
+}
+
+export function editGroupRequest(groupname, saveAndRestoreMode, migrationMode, softwarePackages) {
+  console.log('groupname: ' + groupname + ', saveAndRestoreMode: ' + saveAndRestoreMode + ', migrationMode: ' + migrationMode + ', softwarePackages: ' + softwarePackages);
```
Comment (clairealmozinos) [inline webapp/app/containers/EditGroup/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/EditGroup/actions.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+  return function returnEditGroupRequest(dispatch) {
+    return request
+      .post('http://localhost:8080/api/logged-in/groups/update')
+      .type('form')
+      .send({ groupname, saveAndRestoreMode, migrationMode, softwarePackages })
+      .end((err, res) => {
+        console.log(res.body);
```
Comment (clairealmozinos) [inline webapp/app/containers/EditGroup/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/EditUser/actions.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+export function getUserRequest(username) {
+  return function returnGetUserRequest(dispatch) {
+    return request
+      .get('http://localhost:8080/api/logged-in/admin/users')
+      .end((err, res) => {
+        var i = 0;
+        while (i < res.body.users.length && res.body.users[i].name != username) {
+          ++i;
+        }
+        console.log('User displayed : ' + res.body.users[i].name + ', ' + res.body.users[i].email);
```
Comment (clairealmozinos) [inline webapp/app/containers/EditUser/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/EditUser/actions.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+  };
+}
+
+export function getCurrentUserRequest() {
+  return function returngetCurrentUserRequest(dispatch) {
+    return request
+      .get('http://localhost:8080/api/logged-in/user/profile')
+      .withCredentials()
+      .end((err, res) => {
+        console.log(res.body);
```
Comment (clairealmozinos) [inline webapp/app/containers/EditUser/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/EditUser/actions.js [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+
+export function editUser(user) {
+  return {
+    type: EDIT_USER,
+    user: user,
+  };
+}
+
+export function editUserRequest(username, email, password, confirmation) {
+  console.log('username: ' + username + ', email: ' + email + ', password: ' + password + ', confirmation: ' + confirmation);
```
Comment (clairealmozinos) [inline webapp/app/containers/EditUser/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/EditUser/actions.js [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+  return function returnEditUserRequest(dispatch) {
+    return request
+      .post('http://localhost:8080/api/logged-in/user/profile/update')
+      .type('form')
+      .send({ username, email, password, confirmation })
+      .end((err, res) => {
+        console.log(res.body);
```
Comment (clairealmozinos) [inline webapp/app/containers/EditUser/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/ForgottenPassword/actions.js [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export function forgottenPasswordRequest(email) {
+  return function returnForgottenPasswordRequest(dispatch) {
+    return request
+      .post('http://localhost:8080/api/public/user/recover_password')
+      .type('form')
+      .send({ email })
+      .end((err, res) => {
+        console.log(res.body);
```
Comment (clairealmozinos) [inline webapp/app/containers/ForgottenPassword/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/ForgottenPassword/actions.js [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+        dispatch(forgottenPassword(res.body));
+        if (res.body.success) {
+          dispatch(push('/login'));
+        }
+        else {
+          console.log('Error: invalid or unknown email');
```
Comment (clairealmozinos) [inline webapp/app/containers/ForgottenPassword/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/Groups/actions.js [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+    groups: groups,
+  }
+}
+
+export function getGroupsRequest() {
+  return function returnGetGroupsRequest(dispatch) {
+    return request
+      .get('http://localhost:8080/api/logged-in/admin/groups')
+      .end((err, res) => {
+        console.log(res.body);
```
Comment (clairealmozinos) [inline webapp/app/containers/Groups/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/Login/actions.js [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+export function loginRequest(username, password) {
+  return function returnLoginRequest(dispatch) {
+    return request
+      .post('http://localhost:8080/api/public/user/login/')
+      .type('form')
+      .send({ username, password })
+      .end((err, res) => {
+        console.log(res.body);
```
Comment (clairealmozinos) [inline webapp/app/containers/Login/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/Login/actions.js [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+        dispatch(login(res.body));
+        if (res.body.name) {
+          // dispatch(push('/edit-user'));
+          browserHistory.push('/edit-user/' + username);
+        }
+        else {
+          console.log("Error: invalid username or password");
```
Comment (clairealmozinos) [inline webapp/app/containers/Login/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/Register/actions.js [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+
+export function register(user) {
+  return {
+    type: REGISTER,
+    user: user,
+  };
+}
+
+export function registerRequest(username, email, password, confirmation) {
+  console.log('username: ' + username + ', email: ' + email + ', password: ' + password + ', confirmation: ' + confirmation);
```
Comment (clairealmozinos) [inline webapp/app/containers/Register/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/Register/actions.js [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+  return function returnRegisterRequest(dispatch) {
+    return request
+      .post('http://localhost:8080/api/public/user/sign-up')
+      .type('form')
+      .send({ username, email, password, confirmation })
+      .end((err, res) => {
+        console.log(res.body);
```
Comment (clairealmozinos) [inline webapp/app/containers/Register/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/containers/Register/actions.js [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+        dispatch(register(res.body));
+        if (res.body.success) {
+          dispatch(push('/login'));
+        }
+        else {
+          console.log('Error: register failed');
```
Comment (clairealmozinos) [inline webapp/app/containers/Register/actions.js]: @leroy-p console.log should not be commit.

File: webapp/app/components/Navbar/index.js (hunk 5 above)
Comment (Grenadingue) [inline webapp/app/components/Navbar/index.js:1]: @clairealmozinos should I delete it ? Anything else to delete with that ?

Comment (clairealmozinos) [inline webapp/app/components/Navbar/index.js:1]: No that's all.

**Pull request merged**